Support cold staking

vctt94 opened the pull request:
This PR depends on #2423 

It supports cold staking and is a work towards #1491

vctt94 commented:
Rebased

matheusd commented:
Hmm, I'm currently getting a weird error when starting up a watching only wallet on this PR (blank page and a js console error):
```
fp.js:23 Uncaught TypeError: fn is not a function
    at fp.js:23
    at Array.map (<anonymous>)
    at fp.js:23
    at Connect.configureFinalMapState (connect.js:155)
    at Connect.computeStateProps (connect.js:142)
    at Connect.updateStatePropsIfNeeded (connect.js:204)
    at Connect.render (connect.js:340)
    at finishClassComponent (react-dom.development.js:17184)
    at updateClassComponent (react-dom.development.js:17134)
    at beginWork (react-dom.development.js:18653)
```

This is even after nuking `node_modules` and `app/node_modules` and re-installing via `yarn`.

Are you seeing this as well?

vctt94 commented:
That's probably because of the f5408a5d3172e5c29d28cdaf305dd5a8147425f5 which had some selectors leftovers. I fixed it at 082ca49be8ac2ee84f4b3290936fad74d887754b and now rebased this PR again. 

Can you retry it?
